.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimePeriodExtractor.cs

```diff
@@ -35,6 +35,12 @@ public List<ExtractResult> Extract(string text, DateObject reference)
                 timePeriodErs = TimeZoneUtility.MergeTimeZones(timePeriodErs, config.TimeZoneExtractor.Extract(text, reference), text);
             }
 
+            //TODO: Quick fix to solve german morgen (morning) / morgen (tomorrow) ambiguity, move to config
+            if (text.Equals("morgen"))

```

Comment:
Can you explain how this is solving the problem? I can't follow it from the code only.

Reply:
The problem was that cases like "Morgen Abend" were as "Morning Evening" as "Morgen" can mean both "tomorrow" and "morning". When the extractor extracts "Abend" in the is example it will take the string before that to look for a relative shift to another day like "yesterday", "tomorrow" etc. When if tries to do this on the string "morgen" it will be extracted as a time period ("morning"), not as as "tomorrow". By filtering out "morgen" from the DateTimePeriodExtractor will fix the problem as only in the case where "morgen" is NOT a time period the string "morgen" will be passed to this extractor. I know this isn't a nice fix, but the only one we found in the given time. We already spend days trying to solve this through regex. It should also be solvable through the config but we did not want to introduce changes to the interface and configs for all other languages and did not have more time to work on this.

Reply:
I suggest you create a method called GermanMorgenWorkaround that receives the extractedResults and returns the cleared version if this condition happen. In that method add this description above as comment. Leave the TODO in the current place and add "To be removed after German DateTime v1 is out."
What do you think?